truss/tests/templates/server/test_schema.py

```diff
@@ -0,0 +1,187 @@
+
+    schema = TrussSchema.from_signature(input_signature, output_signature)
+
+    assert schema.input_type == ModelInput
+    assert schema.output_type == ModelOutput
+    assert not schema.supports_streaming
+
+
+def test_truss_schema_non_pydantic_input():
+    def predict(request: str) -> ModelOutput:
+        return ModelOutput(output=request)
+
+    input_signature = inspect.signature(predict).parameters
+    output_signature = inspect.signature(predict).return_annotation
+
+    schema = TrussSchema.from_signature(input_signature, output_signature)
+
+    assert schema is None
+
+
+def test_truss_schema_non_pydantic_output():
+    def predict(request: ModelInput) -> str:
+        return request.input
+
+    input_signature = inspect.signature(predict).parameters
+    output_signature = inspect.signature(predict).return_annotation
+
+    schema = TrussSchema.from_signature(input_signature, output_signature)
+
+    assert schema is None
+
+
+def test_truss_schema_async():
+    async def predict(request: ModelInput) -> Awaitable[ModelOutput]:
+        return ModelOutput(output=request.input)
+
+    input_signature = inspect.signature(predict).parameters
+    output_signature = inspect.signature(predict).return_annotation
+
+    schema = TrussSchema.from_signature(input_signature, output_signature)
+
+    assert schema.input_type == ModelInput
+    assert schema.output_type == ModelOutput
+    assert not schema.supports_streaming
+
+
+def test_truss_schema_streaming():
+    def predict(request: ModelInput) -> Generator[str, None, None]:
+        yield "hello"
+
+    input_signature = inspect.signature(predict).parameters
+    output_signature = inspect.signature(predict).return_annotation
+
+    schema = TrussSchema.from_signature(input_signature, output_signature)
+
+    assert schema.input_type == ModelInput
+    assert schema.output_type is None
+    assert schema.supports_streaming
+
+
+def test_truss_schema_streaming_async():
+    async def predict(request: ModelInput) -> AsyncGenerator[str, None]:
+        yield "hello"
+
+    input_signature = inspect.signature(predict).parameters
+    output_signature = inspect.signature(predict).return_annotation
+
+    schema = TrussSchema.from_signature(input_signature, output_signature)
+
+    assert schema.input_type == ModelInput
+    assert schema.output_type is None
+    assert schema.supports_streaming
+
+
+def test_truss_schema_union_sync():
+    def predict(request: ModelInput) -> Union[ModelOutput, Generator[str, None, None]]:
+        if request.stream:
+            return (yield "hello")
+        else:
+            return ModelOutput(output=request.input)
+
+    input_signature = inspect.signature(predict).parameters
+    output_signature = inspect.signature(predict).return_annotation
+
+    schema = TrussSchema.from_signature(input_signature, output_signature)
+    assert schema.input_type == ModelInput
+    assert schema.output_type is ModelOutput
```

Comment:
super nitty nit: any reason not to use `schema.output_type == ModelOutput` like the previous test cases? I don't have an opinion either way, but it'd be nice to be consistent